velox/type/Variant.h

```diff
@@ -399,8 +399,8 @@ class variant {
   std::string toJson(const TypePtr& type = nullptr) const;
 
   /// Used by python binding, do not change signature.
-  std::string pyToJson() const {
-    return toJson();
+  std::string pyToJson(const TypePtr& type) const {
```

Comment:
Unfortunately @majetideepak this minor change, will break a lot of python code , since the binding will now necessiate a type being passed to use this function. Atleast with c++ we can quantify the places where this change happens , but its far harder in python due to its dynamic nature (duck typing).

Reply:
@kgpai I can understand the complexity. But for correctness, we need the Type since we now have logical types in Velox. For example, a DecimalType is stored as a HUGEINT variant. Now, we cannot print the decimal value back without the type. Does this issue not happen with the Python use cases?
Is the issue that the Type is not present always?
Also, can we use `toJson` itself instead of `pyToJson`?
CC: @mbasmanova

Reply:
Discussed offline with @majetideepak Currently Koski , which uses these python bindings, only supports these types : . 
   ```
   .value("BOOLEAN", velox::TypeKind::BOOLEAN)
      .value("TINYINT", velox::TypeKind::TINYINT)
      .value("SMALLINT", velox::TypeKind::SMALLINT)
      .value("INTEGER", velox::TypeKind::INTEGER)
      .value("BIGINT", velox::TypeKind::BIGINT)
      .value("REAL", velox::TypeKind::REAL)
      .value("DOUBLE", velox::TypeKind::DOUBLE)
      .value("VARCHAR", velox::TypeKind::VARCHAR)
      .value("VARBINARY", velox::TypeKind::VARBINARY)
      .value("TIMESTAMP", velox::TypeKind::TIMESTAMP)
      .value("OPAQUE", velox::TypeKind::OPAQUE)
      .value("ARRAY", velox::TypeKind::ARRAY)
      .value("MAP", velox::TypeKind::MAP)
      .value("ROW", velox::TypeKind::ROW)
     ```
For these types I believe, currently this suffices. Since pyToJson is only used for meta internal use and not in other places - we can keep this as is. 

Again changing this internally would be quite a major change, and can break peoples production jobs/notebooks etc, which I would rather avoid.